functional/pipeline.py

```diff
@@ -1724,3 +1732,64 @@ def _wrap(value):
         return Sequence(value)
     else:
         return value
+
+def extend(func=None, aslist=False, final=False):
+    """
+    Function decorator for adding new methods to the Sequence class.
+
+    >>> @extend
+        def zip2(it):
+            return [(i,i) for i in it]
+
+    >>> seq.range(3).zip2()
+    [(0, 0), (1, 1), (2, 2)]
+
+
+    >>> @extend(aslist=True)
+        def zip2(it):
+            return zip(it,it)
+
+    >>> seq.range(3).zip2()
+    [(0, 0), (1, 1), (2, 2)]
+
+
+    >>> @extend(final=True)
+        def make_set(it):
+            return set(it)
+
+    >>> r = seq([0,1,1]).make_set()
+    >>> r
```

Comment:
I really like the coverage of the various different use cases and capturing all that in one go.

Reply:
Okay. I'll add such tests.